src/main/java/io/axoniq/axonserver/connector/event/EventChannel.java

```diff
@@ -387,4 +390,57 @@ default ResultStream<EventQueryResultEntry> querySnapshotEvents(String queryExpr
     default ResultStream<EventQueryResultEntry> querySnapshotEvents(String queryExpression, boolean liveStream, String contextName) {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Opens the persistent stream identified by {@code streamId}. Fails if the stream does not exist.
+     * @param streamId the unique identification of a persistent stream
+     * @return a PersistedStream streaming events per segment
+     */
+    PersistentStream openPersistentStream(String streamId);
+
+    /**
+     * Opens the persistent stream identified by {@code streamId}. If the stream does not exist it will be
+     * created with the properties specified in {@code creationProperties}.
+     * @param streamId the unique identification of a persistent stream
+     * @param creationProperties properties to initialize the persistent stream if it does not exist yet
+     * @return a PersistedStream streaming events per segment
+     */
+    PersistentStream openPersistentStream(String streamId, PersistedStreamProperties creationProperties);
+
+
+    /**
+     * Deletes a persistent stream.
+     * @param streamId the unique identification of a persistent stream
+     * @return a CompletableFuture that completes when the persistent stream is deleted
+     */
+    CompletableFuture<Void> deletePersistentStream(String streamId);
+
+    /**
+     * Updates the name for a persistent stream.
+     * @param streamId the unique identification of a persistent stream
+     * @param streamName the new logical name for the stream
+     * @return a CompletableFuture that completes when the persistent stream is updated
+     */
+    CompletableFuture<Void> updatePersistentStreamName(String streamId, String streamName);
+
+    /**
+     * Updates the number of segments for a persistent stream.
+     * @param streamId the unique identification of a persistent stream
+     * @param segments the requested number of segments for the persistent stream
+     * @return a CompletableFuture that completes when the persistent stream is updated
+     */
+    CompletableFuture<Void> setPersistentStreamSegments(String streamId, Integer segments);
+
+    /**
+     * Returns a list of persistent streams with the last confirmed token per segment.
+     * @return a list of persistent streams
+     */
+    CompletableFuture<List<StreamStatus>> persistentStreams();
+
+    /**
+     * Returns a list of persistent streams with the connected client per segment.
+     * @return a list of persistent streams
+     */
+    CompletableFuture<List<StreamConnections>> persistentStreamConnections();
```

Comment:
Couldn't this become potentially very long? What about returning the connections for a given streamId? Then the `persistentStreams()` call can be used to retrieve all ID's, and the details can be fetched for those that are relevant.